<?xml version="1.0"?>
<launch>

  <!-- LAUNCH PARAMETERS -->
  <!-- these are the arguments you can pass this launch file, for example paused:=true -->

  <arg name="hardware_interface" default="VelocityJointInterface"/>

  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="false"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  
  
  
  
  
<!-- LAUNCH IMPLEMENTATION -->


<!--Load the Robot URDF and launch Gazebo-->

  <!-- Load an empty world for Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <!-- Load the URDF into the ROS Parameter Server -->
  <param name="robot_description" 
         command="$(find xacro)/xacro.py '$(find yumi_description)/urdf/yumi.urdf.xacro' prefix:=$(arg hardware_interface)" />

  <!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
  <node name="urdf_spawner" 
        pkg="gazebo_ros" 
        type="spawn_model" 
        respawn="false" 
        output="screen"
        args="-urdf -model yumi -param robot_description"/>

  <!-- Run RViz for visualization instead of the Gazebo GUI -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hiqp_controllers)/launch/yumi.rviz"/>


</launch>
